app/websocket/app/main/model_manager.py

```diff
@@ -0,0 +1,195 @@
+import syft as sy
+
+import pickle
+import os
+
+from .persistence.models import db, MLModel
+from sqlalchemy.exc import SQLAlchemyError, IntegrityError
+
+
+model_cache = dict()
+
+
+def clear_cache():
+    """Clears the cache.
+
+    """
+    model_cache = dict()
+
+
+def is_model_in_cache(model_id: str):
+    """Checks if the given model_id is present in cache.
+    
+    Args:
+        model_id (str): Unique id representing the model. 
+
+    Returns:
+        True is present, else False.
+
+    """
+
+    return model_id in model_cache
+
+
+def get_model_from_cache(model_id: str):
+    """Checks the cache for a model. If model not found, returns None.
+    
+    Args:
+        model_id (str): Unique id representing the model. 
+
+    Returns:
+        An encoded model, else returns None.
+
+    """
+
+    return model_cache.get(model_id)
+
+
+def save_model_to_cache(serialized_model: bytes, model_id: str):
+    """Saves the model to cache. Will fail if a model with same id already exists.
+
+    Args:
+        serialized_model (bytes): The model object to be saved.
+        model_id (str): The unique identifier associated with the model.
+
+    """
+    if is_model_in_cache(model_id):
+        return
+    else:
+        model_cache[model_id] = serialized_model
+
+
+def remove_model_from_cache(model_id: str):
+    """Deletes the given model_id from cache.
+    
+    Args:
+        model_id (str): Unique id representing the model. 
+
+    """
+    if is_model_in_cache(model_id):
+        del model_cache[model_id]
+
+
+def list_models():
+    """Returns a list of currently existing models. Will always fetch from db.
+
+    Returns:
+        A dict with structure: {"success": Bool, "models":[model list]}.
+        On error returns dict: {"success": Bool, "error": error message}. 
+
+    """
+
+    try:
+        result = db.session.query(MLModel.id).all()
+        model_ids = [model.id for model in result]
+        return {"success": True, "models": model_ids}
+    except SQLAlchemyError as e:
+        return {"success": False, "error": str(e)}
+
+
+def save_model(serialized_model: bytes, model_id: str):
```

Comment:
@mari-linhares  why not add an update parameter with the save model function in api?
That way if model_id already found... update the model.

Reply:
I think this is a good idea, but this PR is already big enough. We can do this change in a different PR.